salaTest/testmapconversion.cpp

```diff
@@ -0,0 +1,206 @@
+    }
+
+    SECTION("Two lines with extra attributes") {
+        dataMap.makeLineShape(lines[0], false, false, extraAttributes[0]);
+        dataMap.makeLineShape(lines[1], false, false, extraAttributes[1]);
+
+        // TODO: This fails with std::bad_alloc because there's only 2 lines in the data map
+        REQUIRE_THROWS_WITH(MapConverter::convertDataToSegment(nullptr, "Segment map", dataMap, true),
+                            Catch::Contains("std::bad_alloc"));
+    }
+
+    SECTION("Three lines") {
+        dataMap.makeLineShape(lines[0], false, false, extraAttributes[0]);
+        dataMap.makeLineShape(lines[1], false, false, extraAttributes[1]);
+        dataMap.makeLineShape(lines[2], false, false, extraAttributes[2]);
+        std::unique_ptr<ShapeGraph> segmentMap =
+            MapConverter::convertDataToSegment(nullptr, "Segment map", dataMap, true);
+        int segmentNewAttributeID = segmentMap->getAttributeTable().getColumnIndex(newAttributeName);
+        std::map<int, SalaShape> &shapes = segmentMap->getAllShapes();
+        REQUIRE(shapes.size() == 3);
+        auto shapeIter = shapes.begin();
+        for (int i = 0; i < lines.size(); i++) {
+            REQUIRE(shapeIter->first == i);
+            AttributeRow &row = segmentMap->getAttributeRowFromShapeIndex(shapeIter->first);
+            REQUIRE(row.getValue(segmentNewAttributeID) == extraAttributes[i][newAttributeID]);
+            const Line &segmentLine = shapeIter->second.getLine();
+            REQUIRE(segmentLine.ax() == Approx(lines[i].ax()).epsilon(EPSILON));
+            REQUIRE(segmentLine.ay() == Approx(lines[i].ay()).epsilon(EPSILON));
+            REQUIRE(segmentLine.bx() == Approx(lines[i].bx()).epsilon(EPSILON));
+            REQUIRE(segmentLine.by() == Approx(lines[i].by()).epsilon(EPSILON));
+            shapeIter++;
+        }
+    }
+
+    SECTION("Four lines, second line twice") {
+        dataMap.makeLineShape(lines[0], false, false, extraAttributes[0]);
+        dataMap.makeLineShape(lines[1], false, false, extraAttributes[1]);
+        dataMap.makeLineShape(lines[1], false, false, extraAttributes[1]); // this one should be removed by tidylines
+        dataMap.makeLineShape(lines[2], false, false, extraAttributes[2]);
+        std::unique_ptr<ShapeGraph> segmentMap =
+            MapConverter::convertDataToSegment(nullptr, "Segment map", dataMap, true);
+        int segmentNewAttributeID = segmentMap->getAttributeTable().getColumnIndex(newAttributeName);
+        std::map<int, SalaShape> &shapes = segmentMap->getAllShapes();
+        REQUIRE(shapes.size() == 3);
+        auto shapeIter = shapes.begin();
+        for (int i = 0; i < lines.size(); i++) {
+            REQUIRE(shapeIter->first == i);
+            AttributeRow &row = segmentMap->getAttributeRowFromShapeIndex(shapeIter->first);
+            REQUIRE(row.getValue(segmentNewAttributeID) == extraAttributes[i][newAttributeID]);
```

Comment:
Code without this PR fails at this line because the duplicate dataMap line is removed successfully by the conversion but the respective attribute row is not removed along with it.